### PR TITLE
Page/crm about client

### PR DESCRIPTION
**Pull request by YakovchukIvan**
1. Client Page
https://app.clickup.com/t/8692wp6ja

2. Client Page (table)
https://app.clickup.com/t/8692wp6a7

### CODE REVIEW

File: package-lock.json
Comment (pqoqubbw) [inline package-lock.json:1]: we don't use npm, we use yarn, remove this file

File: src/components/Crm-Global-Search/CrmGlobalSeacrh.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { InputAdornment, TextField } from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
+import { seacrhClient } from './style'
+import { useEffect, useState } from 'react'
+
+const CLEAR_TIMER_SEARCH = 2000
+
+type CrmGlobalSearchProps = {
```
Comment (pqoqubbw) [inline src/components/Crm-Global-Search/CrmGlobalSeacrh.tsx]:
```suggestion
type Props = {
```

File: src/components/Crm-Modal-Confirmation/ModalDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  DialogContent,
+  DialogContentText,
+  DialogActions
+} from '@mui/material'
+import InfoIcon from '@mui/icons-material/Info'
+import CloseIcon from '@mui/icons-material/Close'
+
+import { modalDialog, infoIcon, closeIcon, actionsBtn } from './style'
+
+type ModalDialogProps = {
```
Comment (pqoqubbw) [inline src/components/Crm-Modal-Confirmation/ModalDialog.tsx]:
```suggestion
type Props = {
```

File: src/components/crm-grid-table-client/ClientAboutOrdersTable.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import Box from '@mui/material/Box'
+import { DataGrid } from '@mui/x-data-grid'
+import { columns } from './columsGrid'
+
+import { boxTableHistoryOrder, tableHistoryOrder } from './style'
+
+export default function DataGridDemo({ orders = [] }: any) {
```
Comment (pqoqubbw) [inline src/components/crm-grid-table-client/ClientAboutOrdersTable.tsx:7]: do not use `any`

File: src/components/crm-grid-table-client/columsGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    ...statusStyles[status]
+  }
+}
+
+const ORDER_STATUS_TRANSLATIONS = {
+  'new': 'Новий',
+  'in processing': 'В обробці',
+  'shipped': 'Відправлено',
+  'delivered': 'Доставлено',
+  'cancelled': 'Скасовано'
```
Comment (pqoqubbw) [inline src/components/crm-grid-table-client/columsGrid.tsx]: this status should be from be types, @Tkachenko01001  could you show what i mean please

File: src/helpers/formatterTotalPrice.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import { BasketItemsResponse } from '../types'
 
-const calculateTotal = (items: BasketItemsResponse[]) =>
+const calculateTotal = (items: BasketItemsResponse[]) => {
+  console.log('items', items)
```
Comment (pqoqubbw) [inline src/helpers/formatterTotalPrice.ts]: remove console.log

File: src/pages/crm-cleint-about/AboutClient.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import PersonIcon from '@mui/icons-material/Person'
+
+import ChangeRole from './ChangeRole'
+import { useAuth } from '../../hooks/use-auth'
+import { ROLE_TRANSLATIONS } from '../crm-clients-page/CrmTableStickyHead'
+import { Role, UserResponseForCRM } from '../../types'
+import { aboutRole } from './style'
+import InfoClient from './InfoClient'
+
+type AboutClientProps = {
```
Comment (pqoqubbw) [inline src/pages/crm-cleint-about/AboutClient.tsx]:
```suggestion
type Props = {
```

File: src/pages/crm-cleint-about/ChangeRole.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        updated_at: new Date().toISOString()
+      })
+      setUserRole(data.role)
+      setOpenModal(false)
+    } catch (error) {
+      console.error('Помилка зміни ролі:', error)
+    }
+  }
+
+  const changeRole = () => fetchChangeRole(selectedRole)
```
Comment (pqoqubbw) [inline src/pages/crm-cleint-about/ChangeRole.tsx]: why do you need this function, use just `fetchChangeRole(selectedRole)`

File: src/pages/crm-cleint-about/ChangeRole.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+          id="basic-button"
+          aria-controls={open ? 'basic-menu' : undefined}
+          aria-haspopup="true"
+          aria-expanded={open ? 'true' : undefined}
+          onClick={handleClick}
+        >
+          <Typography variant="caption">Змінити роль</Typography>
+          <KeyboardArrowDownIcon fontSize="large" />
+        </Button>
+        <Menu
```
Comment (pqoqubbw) [inline src/pages/crm-cleint-about/ChangeRole.tsx:67]: why do you use `menu` here? it is `select` component or i missed something?

File: src/pages/crm-cleint-about/ConfirmModal.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import ModalDialog from '../../components/Crm-Modal-Confirmation/ModalDialog'
+import { Button } from '../../components/UI/Button'
+import { UserResponseForCRM } from '../../types'
+
+type ConfirmModalProps = {
```
Comment (pqoqubbw) [inline src/pages/crm-cleint-about/ConfirmModal.tsx]:
```suggestion
type Props = {
```

File: src/pages/crm-cleint-about/HistoryOrdersCliet.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        const { data } = await axiosInstance.get<any>(
+          `${BASE_URL_ORDER_CLIENT}limit=${CLIENT_QUANTITY}&offset=${offset}&user_id=${clientId}`
+        )
+
+        const totalQuantityPages = Math.ceil(data.total_count / CLIENT_QUANTITY)
+
+        setPageQty(totalQuantityPages)
+        setPage(nowPage)
+        setOrderHistory(data)
+        setIsLoading(false)
```
Comment (pqoqubbw) [inline src/pages/crm-cleint-about/HistoryOrdersCliet.tsx]: you can do this in `finally`

File: src/pages/crm-cleint-about/InfoClient.tsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import { Box, Typography } from '@mui/material'
+
+type InfoRowProps = {
```
Comment (pqoqubbw) [inline src/pages/crm-cleint-about/InfoClient.tsx]:
```suggestion
type Props = {
```

File: src/pages/crm-clients-page/CrmClientsPage.tsx [hunk 12]
```diff
@@ -1,79 +1,75 @@
...
-        const totalNumberOfPages = Math.ceil(
+        const totalQuantityPages = Math.ceil(
           data.total_count_users / CLIENT_QUANTITY
         )
 
-        setPageQty(totalNumberOfPages)
+        setPageQty(totalQuantityPages)
         setClients(data.users)
+        setPage(nowPage)
+        setIsLoading(false)
```
Comment (pqoqubbw) [inline src/pages/crm-clients-page/CrmClientsPage.tsx]: same here

File: src/pages/crm-orders-page/index.tsx [hunk 13]
```diff
@@ -33,6 +35,8 @@ const CrmOrdersPage = () => {
 
         setPageQty(totalNumberOfPages)
         setOrders(data.orders)
+        // setOrders(data.orders)
```
Comment (pqoqubbw) [inline src/pages/crm-orders-page/index.tsx]: remove unused code

File: src/pages/crm-orders-page/index.tsx [hunk 14]
```diff
@@ -33,6 +35,8 @@ const CrmOrdersPage = () => {
...
+        setIsLoading(false)
```
Comment (pqoqubbw) [inline src/pages/crm-orders-page/index.tsx]: same here

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/components/crm-grid-table-client/columsGrid.tsx (hunk 4 above)
Comment (Tkachenko01001) [reply src/components/crm-grid-table-client/columsGrid.tsx]: code

File: package-lock.json
Comment (YakovchukIvan) [reply package-lock.json:1]: Accidentally... fixed

File: src/components/Crm-Global-Search/CrmGlobalSeacrh.tsx (hunk 1 above)
Comment (YakovchukIvan) [reply src/components/Crm-Global-Search/CrmGlobalSeacrh.tsx]: Done

File: src/components/Crm-Modal-Confirmation/ModalDialog.tsx (hunk 2 above)
Comment (YakovchukIvan) [reply src/components/Crm-Modal-Confirmation/ModalDialog.tsx]: Done

File: src/pages/crm-orders-page/index.tsx (hunk 14 above)
Comment (YakovchukIvan) [reply src/pages/crm-orders-page/index.tsx]: Done

File: src/pages/crm-orders-page/index.tsx (hunk 13 above)
Comment (YakovchukIvan) [reply src/pages/crm-orders-page/index.tsx]: Done

File: src/pages/crm-cleint-about/InfoClient.tsx (hunk 11 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/InfoClient.tsx]: Done

File: src/pages/crm-cleint-about/HistoryOrdersCliet.tsx (hunk 10 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/HistoryOrdersCliet.tsx]: Done

File: src/pages/crm-cleint-about/InfoClient.tsx (hunk 11 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/InfoClient.tsx:3]: Done

File: src/pages/crm-clients-page/CrmClientsPage.tsx (hunk 12 above)
Comment (YakovchukIvan) [reply src/pages/crm-clients-page/CrmClientsPage.tsx]: Done

File: src/components/crm-grid-table-client/columsGrid.tsx (hunk 4 above)
Comment (YakovchukIvan) [reply src/components/crm-grid-table-client/columsGrid.tsx]: Done

**Comment by pqoqubbw** [comment]
fix please deploy error + conflicts

**Comment by pqoqubbw** [comment]
also, could you add ticket url?

File: src/pages/crm-cleint-about/AboutClient.tsx (hunk 6 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/AboutClient.tsx]: Done

File: src/pages/crm-cleint-about/ChangeRole.tsx (hunk 7 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/ChangeRole.tsx]: Done

File: src/pages/crm-cleint-about/ConfirmModal.tsx (hunk 9 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/ConfirmModal.tsx]: Done

File: src/pages/crm-cleint-about/ChangeRole.tsx (hunk 8 above)
Comment (YakovchukIvan) [reply src/pages/crm-cleint-about/ChangeRole.tsx:67]: Done removed aria attributes

**Comment by pqoqubbw** [comment]
updates?

**Comment by pqoqubbw** [comment]
resolve conflicts please + fix issues in "Client Page (table)" task

**Comment by YakovchukIvan** [comment]
Done

**Pull request merged**